teamscale_precommit_client/git_utils.py

```diff
@@ -95,6 +102,20 @@ def get_deleted_files(path_to_repository):
         Returns:
             List(str): List of filenames of all deleted files in the provided repository.
     """
-    repo = Repository(path_to_repository)
-    status_entries = repo.status()
-    return [path for path, st in status_entries.items() if st & _STATI_DELETED]
+    diff = _get_diff_to_last_commit(path_to_repository)
+    return [item.b_path for item in diff if item.change_type == _CHANGE_TYPE_DELETED]
+
+
+def _get_diff_to_last_commit(path_to_repository):
+    """ Utility method for getting a diff between the working copy and the HEAD commit
+
+        Args:
+            path_to_repository (str): Path to the Git repository
+
+        Returns:
+            List(git.diff.Diff): List of Diff objects for every file
+    """
+    repo = Repo(path_to_repository)
+    modified_diff = repo.index.diff(None)
```

Comment:
is this the "unstaged diff" (in contrast to the staged diff?) If so, please name this `unstaged_diff` as the changes in the staging area are also modifications, so "modified" is ambiguous